ColorCycle.cpp

```diff
@@ -0,0 +1,48 @@
+#include "ColorCycle.h"
+
+ColorCycle::ColorCycle() {}
+
+void ColorCycle::start(Adafruit_NeoPixel * neoPixels) {
+  /* check if timeout has expired */
+  if (millis() - timestamp < timeout) {
+    /* timeout hasn't expired, return */
+    return;
+  }
+
+  /* tick the pixel position and color wheel values */
+  tickValues(neoPixels);
+
+  /* get the current color */
+  uint32_t currentColor = colorWheel(wheelValue, neoPixels);
+
+  /* update the pixels */
+  for (uint8_t i = 0; i < neoPixels->numPixels(); i++) {
+    neoPixels->setPixelColor(i, currentColor);
+  }
+
+  neoPixels->show();
+
+  /* update the timestamp */
+  timestamp = millis();
+}
+
+void ColorCycle::tickValues(Adafruit_NeoPixel * neoPixels) {
+  /* increment the color on the color wheel */
+  wheelValue += increment;
+
+  /* reset the value if needed */
+  wheelValue > 255 ? 0 : wheelValue;
+}
+
+uint32_t ColorCycle::colorWheel(byte wheelPosition, Adafruit_NeoPixel * neoPixels) {
```

Comment:
I've extracted the color wheel logic into a new class "Color" with a static method to reduce repeat color wheel logic. Just use `Color.wheel(...)` instead.

Reply:
Nice, thanks! I updated to using the static method in the Color class. Also changed Color.wheel to Color::wheel.